videointelligence/annotate/logo_detection.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package annotate contains speech examples.
+package annotate
+
+// [START video_detect_logo]
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	video "cloud.google.com/go/videointelligence/apiv1"
+	"github.com/golang/protobuf/ptypes"
+	videopb "google.golang.org/genproto/googleapis/cloud/videointelligence/v1"
+)
+
+// logoDetection analyzes a video and extracts logos with their bounding boxes.
+func logoDetection(w io.Writer, filename string) error {
+	// filename := "resources/googlework_tiny.mp4"
+
+	ctx := context.Background()
+
+	// Creates a client.
+	client, err := video.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("video.NewClient: %v", err)
+	}
```

Comment:
defer client.Close() after the err check to clean up the instantiated client, here and in the other sample.